build/scan-appdir.js

```diff
@@ -0,0 +1,91 @@
+'use strict';
+
+/* global require, exports, dump */
+const utils = require('./utils');
+
+var ScanAppdir = function() {
+  var GAIA_APP_CONFIG = utils.getEnv('GAIA_APP_CONFIG');
+  var GAIA_DIR = utils.getEnv('GAIA_DIR');
+  var GAIA_DISTRIBUTION_DIR = utils.getEnv('GAIA_DISTRIBUTION_DIR');
+  var GAIA_APP_SRCDIRS = utils.getEnv('GAIA_APP_SRCDIRS');
+  var appListFile = utils.getFile(GAIA_APP_CONFIG);
+  this.appList = utils.getFileContent(appListFile.exists() ?
+    appListFile : utils.getFile(GAIA_APP_SRCDIRS));
+
+  this.gaiaDirPath = GAIA_DIR;
+  this.gaiaDistributionPath = GAIA_DISTRIBUTION_DIR;
+  this.allAppList = [];
+};
+
+ScanAppdir.prototype.scanAppAndDistributionFolder = function(src, recursive) {
+  var gaiaAppFolder = utils.getFile(this.gaiaDirPath, src);
+  var distributionAppFolder;
+  try {
+    distributionAppFolder = utils.getFile(this.gaiaDistributionPath, src);
+  } catch (e) {
+    distributionAppFolder = null;
+  }
+  var srcAppFolder;
+  if (recursive) {
+    utils.ls(distributionAppFolder, false).forEach(function(subFolder) {
+      this.pushPathSrc(subFolder);
+    }, this);
+    utils.ls(gaiaAppFolder, false).forEach(function(subFolder) {
+      this.pushPathSrc(subFolder);
+    }, this);
+    try {
+      srcAppFolder = utils.getFile(src);
+      utils.ls(srcAppFolder, false).forEach(function(subFolder) {
+        this.pushPathSrc(subFolder);
+      }, this);
+    } catch (e) {}
```

Comment:
Add a comment to remind that we should throw an error after Makefile is replaced with JS module.